editor/create_dialog.cpp

```diff
@@ -351,13 +347,13 @@ float CreateDialog::_score_type(const String &p_type, const String &p_search) co
 	float score = (pos > -1) ? 1.0f - w * MIN(1, 3 * pos * inverse_length) : MAX(0.f, .9f - w);
 
```

Comment:
Consider adding this to the top of `_score_type`, just in case there is an edge case where the bug described in #66198 is still reproducible. (Don't use GitHub to apply this change, it is on the wrong line.)

Am I correct in assuming it is impossible for a search to otherwise have a perfect `1.0f` score?

```suggestion
	if (p_type == p_search) {
		// Always favor an exact match (case-sensitive), since clicking a favorite will set the search text to the type.
		return 1.0f;
	}

```

Reply:
Sound like a good compromise between our two PRs - this addition makes the functionality very clear.
And yes, score is always supposed to be <= 1, as far as I can tell.